src/coffee/sync/utils/product.coffee

```diff
@@ -95,7 +100,57 @@ class ProductUtils extends BaseUtils
       actions.push action if action
     actions
 
-  # Private: map categoryOrderHints actions
+  matchesByIdOrKeyOrSku: (variant1, variant2) ->
+    variant1 and variant2 and (
+      (!isNil(variant1.id) and variant1.id == variant2.id) or
+      (!isNil(variant1.key) and variant1.key == variant2.key) or
+      (!isNil(variant1.sku) and variant1.sku == variant2.sku)
```

Comment:
this `isNil` threw me a little, is it not enough to do 
`variant1.id === variant2.id` 
or 
`variant1.id and variant1.id === variant2.id)`

?

Reply:
If we don't use `isNil`, it would match incorrectly variants where existing and the new variant id is undefined even when skus and keys are different.

If we change it to `variant1.id and variant1.id === variant2.id` it won't work when id is `0`.

Reply:
ahh.. I see. Thanks for explaining.